src/services/blocks/BlocksService.ts

```diff
@@ -524,6 +536,32 @@ export class BlocksService extends AbstractService {
 		};
 	}
 
+	/**
+	 * The block where the runtime is deployed falsely proclaims it would
```

Comment:
```suggestion
	 * The block where the runtime is enacted falsely proclaims it would
```

…or do I misunderstand. This is about the block where a new runtime is activated?

Reply:
I understand deployed and enacted as synonymous here: It is defined as the block that contains the `set_code` dispatchable that sets the new runtime code. The point is that when you query for the runtime version at that block via RPC you get the just newly deployed version. This is because on RPC a new wasm instance of the current code is created, which is the just deployed one.

However, block production still happens with the old instance which means that the block after the one with `set_code` is created with the old runtime. This is why we need this hack.

Reply:
When I first wrote my comment I thought the new blob was already present on chain and activated at the given block. I later learned I was wrong, and your comment further clarifies things, ty. So "deployed" here is fine.

Reply:
Thanks for the great explanation!